File: Rotate_Image/rotate_image.cpp
#include <vector>
using namespace std;

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] =  temp;
            }
        }

        for (int x = 0; x < n; x++) {
            for (int y = 0; y < n / 2; y++) {
                int temp = matrix[x][y];

                matrix[x][y] = matrix[x][n - 1 - y];
                matrix[x][n - 1 - y] = temp;
            }
        }

    }
};

File: Rotate_Image/rotate_image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        # transpose it, and then reverse each row.

        n = len(matrix)
        for i in range(n):
            for j in range(i + 1, n):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
        
        for x in range(n):
            for y in range(n // 2):
                matrix[x][y], matrix[x][-1 - y] = matrix[x][-1 - y], matrix[x][y]



        return None